File: C++/Solution119.cpp
#include<iostream>
#include<vector>
#include<algorithm>
#include<unordered_map>
#include<unordered_set>
#include<queue>
using namespace std;

typedef long long ll;
typedef pair<int, int> pii;


class Solution119 {
public:
    vector<int> getRow(int rowIndex) {
        auto vec = generate(rowIndex + 1);
        return vec.back();
    }

    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ans;
        ans.push_back({1});
        while (numRows > 1) {
            auto &last = ans.back();
            vector<int> cur;
            cur.push_back(1);
            for (int i = 0; i + 1 < last.size(); i++)
            {
                cur.push_back(last[i] + last[i + 1]);
            }
            cur.push_back(1);
            ans.push_back(cur);
            numRows--;
        }
        return ans;
    }
};

File: Java/other/Solution119.java
import java.util.List;
import java.util.LinkedList;

class Solution119 {
	public List<Integer> getRow(int rowIndex) {
		List<Integer> res = new LinkedList<>();
		long cur = 1;
		for (int i = 0; i <= rowIndex; i++) {
			res.add((int) cur);
			cur = cur * (rowIndex-i)/(i+1);
		}
		return res;
	}

    public static void main(String[] args){
    	Solution119 s = new Solution119();
    	System.out.println(s.getRow(13));

    }
}